### PR TITLE
pista: init at v0.1.5

### PR DESCRIPTION
**Pull request by oppiliappan**
###### Motivation for this change

Adds the `pista` prompt to nixpkgs.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

This is my first contribution to `nixpkgs`, I hope everything is in order.

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118914` at e8f96d47 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/3e2d5cc6fdaab543fd278da3fabfefa9">pista</a></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `stdenv`.
Near pkgs/tools/misc/pista/default.nix:1:3:
```
  |
1 | { stdenv, lib, rustPlatform, fetchFromGitHub, openssl, pkg-config }:
  |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -6825,6 +6825,12 @@
     githubId = 628342;
     name = "Tim Steinbach";
   };
+  nerdypepper = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:6828]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/tools/misc/pista/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  };
+  cargoSha256 = "1qjg4z4nfqmiawczs4p4rf4mq250aqb5brd3hd2jcjdhqajksr90";
+
+  meta = with lib; {
+    description = "Simple {bash, zsh} prompt for programmers";
+    homepage = "https://github.com/NerdyPepper/pista";
+    license = licenses.mit;
+    maintainers = with maintainers; [ nerdypepper ];
+  };
+}
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/pista/default.nix:23]:
```suggestion
{ stdenv, lib, rustPlatform, fetchFromGitHub, openssl, pkg-config }:

rustPlatform.buildRustPackage rec {
  pname = "pista";
  version = "0.1.5";

  src = fetchFromGitHub {
    owner = "NerdyPepper";
    repo = "pista";
    rev = "v${version}";
    sha256 = "1nrqgl5r5ja95g5xhfvkvfz72g318pqc7yj4fwl6xnad19smvms0";
  };

  cargoSha256 = "1qjg4z4nfqmiawczs4p4rf4mq250aqb5brd3hd2jcjdhqajksr90";

  nativeBuildInputs = [ pkg-config ];
  buildInputs = [ openssl ];

  # has not tests
  doCheck = false;

  meta = with lib; {
    description = "Simple {bash, zsh} prompt for programmers";
    homepage = "https://github.com/NerdyPepper/pista";
    license = licenses.mit;
    maintainers = with maintainers; [ nerdypepper ];
  };
}
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by NerdyPepper** [comment]
Made changes as requested.

**Comment by risicle** [comment]
Um.. https://github.com/NerdyPepper/pista/archive/v0.1.5.tar.gz appears to have disappeared and is now 404ing for me.

**Comment by NerdyPepper** [comment]
@risicle should be working now.

**Comment by SuperSandro2000** [comment]
@ofborg eval
@ofborg build

File: pkgs/tools/misc/pista/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+rustPlatform.buildRustPackage rec {
+  pname = "pista";
+  version = "0.1.5";
+
+  src = fetchFromGitHub {
+    owner = "NerdyPepper";
+    repo = "pista";
+    rev = "v${version}";
+    sha256 = "1nrqgl5r5ja95g5xhfvkvfz72g318pqc7yj4fwl6xnad19smvms0";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/pista/default.nix:11]:
```suggestion
    sha256 = "0fbvi85vlh7pk0f6d2q05na2jpbgwpylwzzms6syvywk8dm9hd1f";
```

Comment (NerdyPepper) [reply pkgs/tools/misc/pista/default.nix:11]: Seems to break the build. Setting `fetchSubmodules` to `true` works though. Not sure how the build succeeded previously.

**Comment by roberth** [comment]
lgtm. Please rebase and test.

**Comment by SuperSandro2000** [comment]
Closing due to inactivity from author.
